pkg/collectors/route.go

```diff
@@ -86,7 +91,16 @@ var (
 								m.RouterName,
 							},
 							Value: 1,
-						})
+						}
+						// Avoid duplicates: see OCPBUGS-48747 for more information.
+						if len(f.Metrics) > 0 {
+							previousMetric := f.Metrics[len(f.Metrics)-1]
```

Comment:
do we have a guarantee that only the previous condition could be a "duplicate"?

Reply:
You mean if it's like a ABA type of arrangement in the `v1.Conditions` field? I assumed they'd define a [`listMapKey`](https://github.com/rexagod/resource-state-metrics/blob/main/pkg/apis/resourcestatemetrics/v1alpha1/types.go#L93) to organize but that does not seem to be the case, I'll refactor this.

Reply:
The conclusion was that 
```
  status:
    ingress:
    - conditions:
      - lastTransitionTime: "XXXX"
        status: "True"
        type: Admitted
      - message: foo
        reason: MyReason
        status: "True"
        type: Admitted
```

isn't expected (we should have at most one `type: Admitted`).

So there is no dedup to do here.

Let's just add a comment somewhere that `type, status` is guaranteed to be unique for each `job, router_name`.

Reply:
Thanks for checking this with the network team, I've reverted the change and added a comment.

Reply:
Sure, so we don't need to do any deduplication in here, we want Prometheus to continue complaining and warning us in the future if the conditions are not well formed.

let's just add a comment that the current code in main expects type: Admitted to be unique